docs/DeveloperGuide.md

```diff
@@ -25,6 +40,159 @@
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
 |v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
 
+### Main Menu
+This section describes the implementation of core main menu features.
+
+**API:** `AllOnUs.java`
+
+The menu is accessed through a call to static method main() of the AllOnUs class. 
+The Class Diagram below shows the overall structure of the application from the menu's point of view.
+The AllOnUs class and other classes that are coupled to the menu are therefore included.
+
+![](images/MenuFeaturesClassDiagram.png)
+
+Note: Exception classes are left out of this diagram that aims to show the core structure of the application. is*Command()
+method here refers to all methods of this form (more on this will be elaborated later in detail). 
+
+As illustrated in the Class Diagram above, the AllOnUs class only associates to one of each of the other classes.
+These classes are, namely, the ContactsManager class which manages the contacts section of the application, 
+the StudyManager class which manages the module and timetable part of the application, the ExpenseTracker class
+which manages the expenses section of the application, the Logger class which helps with logging information about the 
+state of the programme when it is running and the TextUi class which handles user input from command line.
+
+Now we will look at the sequence of operations that take place upon a call to method main() of the AllOnUs class. 
+Below is a sequence diagram that shows the core interactions between associated classes and objects in the execution 
+and runtime of the programme.
+
+![](images/MenuFeaturesSequenceDiagram.png)
+
+![](images/MenuFeaturesSequenceSubDiagram.png)
+
+As illustrated in the diagram, once main() method of AllOnUs is called, a new unnamed :AllOnUs object is created 
+for which we execute the non-static method run(), which contains a loop for sustained interaction with the user
+(command line). Objects for the ContactsManager (contactsManager), StudyManager (studyManager) and ExpenseTracker (expenseTracker)
+classes are also created, where Logger (logger) and TextUi (ui) objects are already contained in the fields of the 
+:AllOnUs object. 
+
+In the loop, run() method calls the getUserInput() method, which belongs to ui, and then checks the returned string for 
+whether it is a command to enter a particular section, to get help, to exit the application or is an empty or unrecognizable 
+command. 
+
+Command to enter a particular section: 
+The returned string from getUserInput() call becomes a parameter for a function call to static method is*Command(), which
+belongs to the AllOnUs class, and the is*Command() method is of the form isStudyManagerCommand(), isContactsManagerCommand(),
```

Comment:
Perhaps the asterisk in `is*Command()` can be escaped? I noticed that the asterisks are being interpreted as italics.

Screenshot of example:
![image](https://user-images.githubusercontent.com/30099983/160958883-8d504c23-cf27-45f0-9f2a-40f0254a7cf3.png)


Alternatively, you could mark up the `is*Command()` methods :)